cli/dcoscli/node/main.py

```diff
@@ -748,20 +750,25 @@ def _ssh(leader, slave, option, config_file, user, master_proxy, command):
     if command is None:
         command = ''
 
-    master_public_ip = dcos_client.metadata().get('PUBLIC_IPV4')
-    if master_proxy:
+    proxy_ip = proxy_ip or config.get_config_val('core.dcos_proxy_ip')
+    if proxy_ip:
+        master_public_ip = proxy_ip
```

Comment:
This assignment is not correct. Lets use a different variable to reference what we use to "proxy". `proxy_ip = master_public_ip` would make sense here. Then we can also update the references to master_public_ip and proxy_ip below.